docs/core/porting/upgrade-assistant-wcf.md

```diff
@@ -0,0 +1,420 @@
+
+```
+9. Update source code
+    a. Apply fix for UA0002: Types should be upgraded
+    b. Apply fix for UA0012: 'UnsafeDeserialize()' does not exist
+    c. Apply fix for UA0014: .NET MAUI projects should not reference Xamarin.Forms namespaces
+    d. Apply fix for UA0015: .NET MAUI projects should not reference Xamarin.Essentials namespaces
+```
+
+In this case, none of the suggested fixes apply to the example project, and this step automatically completes immediately after the previous step was completed.
+
+```
+[10:26:44 INF] Initializing upgrade step Update source code
+[10:26:44 INF] Running analyzers on CalculatorService
+[10:26:48 INF] Identified 0 diagnostics in project CalculatorService
+[10:26:51 INF] Initializing upgrade step Move to next project
+```
+
+#### Completing the upgrade
+
+If there are any more projects to migrate, the tool lets you select which project to upgrade next. When there are no more projects to upgrade, the tool brings you to the "Finalize upgrade" step:
+
+```
+1. [Next step] Finalize upgrade
+
+Choose a command:
+   1. Apply next step (Finalize upgrade)
+   2. Skip next step (Finalize upgrade)
+   3. See more step details
+   4. Configure logging
+   5. Exit
+>
+[10:27:15 INF] Applying upgrade step Finalize upgrade
+[10:27:15 INF] Upgrade step Finalize upgrade applied successfully
+```
+
+Ideally, after successfully running the tool, these modified files should look similar. Please check out the updated version of [CalculatorService.csproj](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/CalculatorService.csproj), [app.config](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/App.config), [wcf.config](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/wcf.config), [CalculatorSample/service.cs](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/service.cs).
+
+Notice that if you want to run both CalculatorClient and CalculatorService, you would need to upgrade CalculatorClient with the tool as well. Because CalculatorService targets .Net6.0 after upgrade, it cannot be referenced by CalculatorClient which still targets .NETFramework.
+
+## After upgrading
+
+After you upgrade your projects, you'll need to compile and test them. Most certainly you'll have more work to do in finishing the upgrade. It's possible that the .NET Framework version of your app contained library references that your project isn't actually using. You'll need to analyze each reference and determine whether or not it's required. The tool may have also added or upgraded a NuGet package reference to wrong version.
```

Comment:
```suggestion
After you upgrade your projects, you'll need to compile and test them. The upgrade assistant will do what it can, but it can't solve every incompatibility as part of the project upgrade. For example, it's possible that the .NET Framework version of your app contained library references that your project isn't actually using. You'll need to analyze each reference and determine whether or not it's required. The tool may have also added or upgraded a NuGet package reference to wrong version.
```